project/AkkaBuild.scala

```diff
@@ -1382,16 +1385,43 @@ object AkkaBuild extends Build {
 
     val cluster = exports(Seq("akka.cluster.*"), imports = Seq(protobufImport()))
 
-    val parsing = exports(Seq("akka.parboiled2.*", "akka.shapeless.*"))
+    val parsing = exports(Seq("akka.parboiled2.*", "akka.shapeless.*"),
+        imports = Seq(optionalResolution("scala.quasiquotes")))
+
+    val httpCore = exports(Seq("akka.http.*"),
+        imports = Seq(streamAndHttpImport("akka.stream.*"),
+            streamAndHttpImport("akka.parboiled2.*"),
+            streamAndHttpImport("akka.shapeless.*")))
+
+    val http = exports(Seq("akka.http.impl.server",
+        "akka.http.scaladsl.server.*", "akka.http.javadsl.server.*",
+        "akka.http.scaladsl.client", "akka.http.scaladsl.coding", "akka.http.scaladsl.common",
+        "akka.http.scaladsl.marshalling", "akka.http.scaladsl.unmarshalling"),
+        imports = Seq(streamAndHttpImport("akka.stream.*"),
+            streamAndHttpImport("akka.http.*"),
+            streamAndHttpImport("akka.parboiled2.*")))
 
-    val httpCore = Nil // FIXME #15689
+    val httpTestKit = exports(Seq("akka.http.scaladsl.testkit.*", "akka.http.javaadsl.testkit.*"),
```

Comment:
Hmm, shouldn't the script written by @2m have detected this problem?

Also, the verification script should check that the generated jar does not contain _more_ classes than `target/classes` folder, because it's easy to accidentally re-export packages from an upstream dependency if a too broad Export declaration is used (which is completely valid, and occasionally useful in OSGi environment)